core/src/main/java/eu/tsystems/mms/tic/testframework/execution/testng/SimpleAssertion.java

```diff
@@ -29,156 +29,208 @@
         assertContainsNot(actual, expected, null);
     }
 
     void assertGreaterThan(BigDecimal actual, BigDecimal expected, Object subject);
+
     default void assertGreaterThan(BigDecimal actual, BigDecimal expected) {
         assertGreaterThan(actual, expected, null);
     }
+
     default void assertGreaterThan(Object actual, Object expected, Object subject) {
         assertGreaterThan(toBigDecimal(actual), toBigDecimal(expected), subject);
     }
+
     default void assertGreaterThan(Object actual, Object expected) {
         assertGreaterThan(actual, expected, null);
     }
 
 
     void assertGreaterEqualThan(BigDecimal actual, BigDecimal expected, Object subject);
+
     default void assertGreaterEqualThan(BigDecimal actual, BigDecimal expected) {
         assertGreaterEqualThan(actual, expected, null);
     }
+
     default void assertGreaterEqualThan(Object actual, Object expected) {
         assertGreaterEqualThan(actual, expected, null);
     }
+
     default void assertGreaterEqualThan(Object actual, Object expected, Object subject) {
         assertGreaterEqualThan(toBigDecimal(actual), toBigDecimal(expected), subject);
     }
 
     void assertLowerThan(BigDecimal actual, BigDecimal expected, Object subject);
+
     default void assertLowerThan(BigDecimal actual, BigDecimal expected) {
         assertLowerThan(actual, expected, null);
     }
+
     default void assertLowerThan(Object actual, Object expected, Object subject) {
         assertLowerThan(toBigDecimal(actual), toBigDecimal(expected), subject);
     }
+
     default void assertLowerThan(Object actual, Object expected) {
         assertLowerThan(actual, expected, null);
     }
 
     void assertLowerEqualThan(BigDecimal actual, BigDecimal expected, Object subject);
+
     default void assertLowerEqualThan(BigDecimal actual, BigDecimal expected) {
         assertLowerEqualThan(actual, expected, null);
     }
+
     default void assertLowerEqualThan(Object actual, Object expected) {
         assertLowerEqualThan(actual, expected, null);
     }
+
     default void assertLowerEqualThan(Object actual, Object expected, Object subject) {
         assertLowerEqualThan(this.toBigDecimal(actual), this.toBigDecimal(expected), subject);
     }
 
     void assertBetween(BigDecimal actual, BigDecimal lower, BigDecimal higher, Object subject);
+
     default void assertBetween(BigDecimal actual, BigDecimal lower, BigDecimal higher) {
         assertBetween(actual, lower, higher, null);
     }
+
     default void assertBetween(Object actual, Object lower, Object higher) {
         assertBetween(actual, lower, higher, null);
     }
+
     default void assertBetween(Object actual, Object lower, Object higher, Object subject) {
         assertBetween(toBigDecimal(actual), toBigDecimal(lower), toBigDecimal(higher), subject);
     }
 
     void assertEquals(Object actual, Object expected, Object subject);
+
+    void assertEquals(long actual, long expected, Object subject);
```

Comment:
I'm confused. We already have a default method `default void assertEquals(long actual, long expected)`

Reply:
I've rechecked the behaviour. There is a difference using an equal between float and double:
```java
double c = 1.0;
float d = 1.0F;
Assert.assertEquals(c, d); // --> pass
ASSERT.assertEquals(c, d); // --> fail
```

Reply:
While `ASSERT.assertEquals(a, b);` for one long and one int value would pass, adding a third argument (a subject) would cause the assertion to fail: `ASSERT.assertEquals(a, b, "Test");` which has been fixed.